main.tf

```diff
@@ -1,152 +1,165 @@
-  compartment_id = var.compartment_id
-  label_prefix   = var.label_prefix
-  ad_names       = module.base.ad_names
-
-  # network parameters
-  ig_route_id  = module.base.ig_route_id
-  nat_route_id = module.base.nat_route_id
-  netnum       = var.netnum
-  newbits      = var.newbits
-  vcn_id       = module.base.vcn_id
+resource "null_resource" "ocne_config_validation" {
+  depends_on = [data.external.ocne_config]
+  lifecycle {
+    postcondition {
+      condition       = (length(data.external.ocne_config.result.error) == 0)
+      error_message   = "${data.external.ocne_config.result.error}"
+    }
+  }
 }
 
-module "master" {
-  source = "./modules/master"
-
-  # general oci parameters
-  compartment_id = var.compartment_id
-  label_prefix   = var.label_prefix
-  ad_names       = module.base.ad_names
-
-  # networking parameters
-  nsg_id       = module.network.nsg_ids["master"]
-  subnet_id    = module.network.subnet_ids["masters"]
-  subnet_label = module.network.master_subnet_dns_label
-
-  # master compute parameters
-  master_image_id     = var.master_image_id
-  master_shape        = var.master_shape
-  master_size         = var.master_size
-  master_upgrade      = var.master_package_upgrade
-  ssh_public_key      = ""
-  ssh_public_key_path = var.ssh_public_key_path
-  timezone            = var.master_timezone
-
+module "oci-ocne-network" {
+  source     = "./modules/oci-ocne-network"
+  count      = (var.deploy_networking && var.enable_bastion) || var.enable_bastion ? 1 : 0
+
+  tenancy_id           = var.tenancy_id
+  compartment_id       = var.compartment_id
+  region               = var.region
+  fingerprint          = var.fingerprint
+  api_private_key_path = var.api_private_key_path
+  ssh_public_key_path  = var.ssh_public_key_path
+  user_id              = var.user_id
+  prefix               = var.prefix
+  deploy_networking    = var.deploy_networking
+  enable_bastion      = var.enable_bastion
+  vcn_id               = var.vcn_id
+  ig_route_id          = var.ig_route_id
+  nat_route_id         = var.nat_route_id
+  freeform_tags        = var.freeform_tags
 }
 
-module "worker" {
-  source = "./modules/worker"
-
-  # general oci parameters
-  compartment_id = var.compartment_id
-  label_prefix   = var.label_prefix
-  ad_names       = module.base.ad_names
-
-  # networking parameters
-  nsg_id       = module.network.nsg_ids["worker"]
-  subnet_id    = module.network.subnet_ids["workers"]
-  subnet_label = module.network.worker_subnet_dns_label
-
-  # worker compute parameters
-  worker_image_id     = var.worker_image_id
-  worker_shape        = var.worker_shape
-  worker_size         = var.worker_size
-  worker_upgrade      = var.worker_package_upgrade
-  ssh_public_key      = ""
-  ssh_public_key_path = var.ssh_public_key_path
-  timezone            = var.worker_timezone
-
-  # olcne public load balancer to attach to worker pool
-  oci_loadbalancer_id = module.loadbalancer.pub_lb_id
+module "infrastructure" {
```

Comment:
What kind of infrastructure does this module handle? Perhaps a short comment will be useful here